synapse/rest/client/v2_alpha/account.py

```diff
@@ -143,6 +144,10 @@ async def on_POST(self, request):
             # Wrap the session id in a JSON object
             ret = {"sid": sid}
 
+        threepid_send_requests.labels(type="email", reason="password_reset").observe(
+            send_attempt
```

Comment:
It's worth noting that the spec allows client to [put any integer](https://matrix.org/docs/spec/client_server/latest#post-matrix-client-r0-register-email-requesttoken) for `send_attempt`, which may generate incorrect metric results. I'm not sure if known good clients converge on the same behaviour as a result.

Additionally, it looks as though we don't even check if this value is an integer...

Reply:
TBH I think I'm fine with that. I believe the code would have exploded before if the client didn't give it an int since the DB uses an int field.

Reply:
Ah, yes good point, this would be on the other side of the DB.

On the "any integer" bit, we concluded in #synapse-dev that most clients start from 1, which will yield uniform metrics, and some outliers are no big deal.